connection/connection.go

```diff
@@ -188,7 +190,7 @@ func connect(
 			if haveConnected && !lostConnection {
 				// We have detected a loss of connection for the first time. Decide what to do...
 				// Record this once. TODO (?): log at regular time intervals.
-				klog.Errorf("Lost connection to %s.", address)
+				klog.FromContext(ctx).Error(nil, "Lost connection", "address", address)
```

Comment:
In functions with several different log calls, use a `logger := klog.FromContext(ctx)` helper variable.